### ISSUE TITLE
Framerate and Profiler is Per-Scene

### ISSUE THREAD

**Issue by SolarLune** [GoranM/bdx#718]
Hey, there. So basically, if you enable Framerate and Profiler and then disable it in another scene that gets added later, the profiler isn't visible.

Basically, the framerate and profiler should be set by default only once from Blender; apart from that, it should only be altered via code by the user.

**Comment by GoranM** [comment]
It's probably a good idea to summon @rafcolson for anything profiler related.

**Comment by rafcolson** [comment]
>It's probably a good idea to summon @rafcolson for anything profiler related.

I'll take a look at it today. There's also another related issue that should be addressed. When the profiler is enabled in multiple scenes Bdx generates duplicates of the font.

**Comment by rafcolson** [comment]
>if you enable Framerate and Profiler and then disable it in another scene that gets added later, the profiler isn't visible.

Can't reproduce. What I did is enable Framerate and Profile, create an overlay scene (in which the Framerate and Profile is disabled by default) and added the code which adds the overlay scene in `Sacky.init()`. The profiler is visible when running the game.

**Comment by SolarLune** [comment]
> However, when it's the other way around, then the profiler isn't visible. So, when the overlay has Framerate and Profile enabled, but the main scene hasn't.

Hmm, that's not what I'm getting in Gearend - the profiler is visible even though it's disabled in the starting scene. It looks like `put()` (and other functions) call reinitialize() on the Profiler, which means the Profiler can be disabled or enabled by another scene that is created. That might be the problem. 

I guess a way to test it would be to `put()` a property in the props HashMap before adding the overlay scene in `Sacky.init()` in your test file.

**Comment by rafcolson** [comment]
Thanks for that info. This should fix it. However, I also noticed that with current master the profiler's background isn't visible anymore. In any case, this should be fixed with the Viewport commit too.

PR submitted. @GoranM It may be easier for me to adjust this commit after the Viewport commit is already merged. Thanks.

**Comment by rafcolson** [comment]
PR merged. Could this be closed?

**Comment by SolarLune** [comment]
Yeah, okay, sure. Thanks!
